pkg/snapstore/abs_snapstore.go

```diff
@@ -131,7 +142,9 @@ func (a *ABSSnapStore) List() (brtypes.SnapList, error) {
 		// Process the blobs returned in this result segment
 		for _, blob := range listBlob.Segment.BlobItems {
 			if strings.Contains(blob.Name, backupVersionV1) || strings.Contains(blob.Name, backupVersionV2) {
-				s, err := ParseSnapshot(path.Join(prefix, blob.Name))
+				//the blob may contain the full path in its name including the prefix
+				blobName := strings.TrimPrefix(blob.Name, prefix)
+				s, err := ParseSnapshot(path.Join(prefix, blobName))
```

Comment:
@abdasgupta Can you please check if this change is required for the changes for folder structure change also?

Reply:
This is not required for folder structure change

Reply:
@kris94 Why are we trimming of prefix and then joining again? Is it to get rid of any extra `/` ?

Reply:
During the my tests with `v2` I have observed  that the `blobName` comes with the prefix concatenated to it. So if we have `--store-prefix=prefix` then the blobName would be `prefix/Name` and with that `Join` `prefix/prefix/Name`, which points to a snapshot that does not exists. I assumed that this was not the case with `v1` or at leas in some point in the past, so I just put a `TrimPrefix` for backward compatibility.